pkg/apis/componentconfig/types.go

```diff
@@ -0,0 +1,67 @@
+/*
+Copyright 2015 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package componentconfig
+
+// TODO(componentconfig-q): Should these be in pkg/apis/componentconfig?
+// or pkg/apis/auth or pkg/apis/authconfig or ???
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// TODO(componentconfig-q): Is the Auth in AuthConfiguration redundant?
```

Comment:
If Auth is in the name of the API group, it doesn't strictly need to be in the name of the Kind, though we don't currently have good mechanisms to disambiguate non-unique kinds, and "Configuration" would be generic enough to not be very useful.

I'd like components to support consuming multiple config Kinds, perhaps from multiple files/URLs/whatever. In that case, the types could have more specific names, like ProviderList.